### PR TITLE
Fixed namespace validation | url trim in cluster

### PR DESCRIPTION
**Pull request by milindk8**
# Description

1. Fixed & made changes in the regex of namespace validation to allow users to create namespace starting with an alphanumeric character

2. Trim the URL input in cluster & environment < Create cluster

Fixes # [ISSUE 496](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/496), [ISSUE 222](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/222)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

##Testing 

1. Testing is done by creating an environment, and namespace in global configuration
2. By creating a cluster and adding URL with blank spaces

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/config/constants.ts [hunk 1]
```diff
@@ -201,6 +201,7 @@ export const PATTERNS = {
     CONFIG_MAP_AND_SECRET_MULTPLS_KEYS: /^[-._a-zA-Z0-9\,\?\s]*[-._a-zA-Z0-9\s]$/,
     VARIABLE: /^[A-z0-9-_]+$/,
     API_TOKEN: '^[a-z0-9][a-z0-9_-]*[a-z0-9]$/*',
+    NAME_SPACE: '^[a-z0-9][a-z0-9-.]*[a-z0-9]$/*'
```
Comment (sohel-rp) [inline src/config/constants.ts:204]: change to `NAMESPACE`

File: src/components/cluster/Cluster.tsx [hunk 2]
```diff
@@ -788,7 +788,7 @@ function Environment({
                     { error: 'Environment name is required', regex: /^.*$/ },
                     { error: "Use only lowercase alphanumeric characters or '-'", regex: /^[a-z0-9-]+$/ },
                     { error: "Cannot start/end with '-'", regex: /^(?![-]).*[^-]$/ },
-                    { error: 'Minimum 3 and Maximum 16 characters required', regex: /^.{3,16}$/ },
+                    { error: 'Minimum 1 and Maximum 16 characters required', regex: /^.{1,16}$/ },
```
Comment (sohel-rp) [inline src/components/cluster/Cluster.tsx:791]: Is this expected? We have the rule for min 3, right?

Comment (milindk8) [reply src/components/cluster/Cluster.tsx:791]: Yes, Utkarsh & Prakarsh decide that namespace can be a minimum 1 letter

File: src/components/cluster/Cluster.tsx [hunk 3]
```diff
@@ -514,7 +514,7 @@ function ClusterForm({
 
     async function onValidation() {
         let payload = getClusterPayload()
-
+        state.url.value = state.url.value.trim()
```
Comment (sohel-rp) [inline src/components/cluster/Cluster.tsx:517]: Can we have a new var instead of re-assigning the trimmed value?

**Review by vivek-devtron — APPROVED** [review]

File: src/config/constants.ts (hunk 1 above)
Comment (milindk8) [reply src/config/constants.ts]: Done

File: src/components/cluster/Cluster.tsx (hunk 3 above)
Comment (milindk8) [reply src/components/cluster/Cluster.tsx]: Done

File: src/components/cluster/Cluster.tsx (hunk 2 above)
Comment (sohel-rp) [reply src/components/cluster/Cluster.tsx:791]: Okay

**Review by sohel-rp — APPROVED** [review]

**Review by shivani170 — APPROVED** [review]

**Pull request merged**